### PR TITLE
feat(katana): update RPC to v0.5.1

### PR DESCRIPTION
**Pull request by kariy**
Update Katana RPC server to version [0.5.1](https://github.com/starkware-libs/starknet-specs/tree/v0.5.1) of Starknet JSON-RPC specs. The new spec introduces two new read API as well as some updates (additions & removals of fields) on the existing types. The types from the `primitive` crate have also been updated in order to accommodate these changes.

The new read APIs are:

- `starknet_getTransactionStatus`
- `starknet_specVersion`

This update requires a bump of `starknet-rs` to `0.8.0`.

## Checklist

- [x] implement `starknet_getTransactionStatus`
- [x] implement `starknet_specVersion`
- [x] remove `starknet_pendingTransactions`
- [x] update block header types to include gas prices
- [x] update block header types

### CODE REVIEW

**Comment by kariy** [comment]
@glihm do you mind reviewing ?

File: examples/rpc/starknet/starknet_getBlockWithTxs.hurl [hunk 1]
```diff
@@ -12,5 +12,8 @@ Content-Type: application/json
 HTTP 200
 [Asserts]
 jsonpath "$.error" not exists
-# jsonpath "$.result.block_hash" matches /^0x[A-Fa-f0-9]+$/
 jsonpath "$.result.transactions" isCollection
+jsonpath "$.result.block_number" isInteger
+jsonpath "$.result.block_hash" matches /^0x[A-Fa-f0-9]+$/
+jsonpath "$.result.parent_hash" matches /^0x[A-Fa-f0-9]+$/
+jsonpath "$.result.starknet_version" matches /^[0-9]+.[0-9]+.[0-9]+$/
```
Comment (glihm) [inline examples/rpc/starknet/starknet_getBlockWithTxs.hurl:19]:
```suggestion
jsonpath "$.result.starknet_version" matches /^[0-9]+.[0-9]+.[0-9]+$/

```

File: examples/rpc/starknet/starknet_getStorageAt.hurl [hunk 2]
```diff
@@ -1,17 +1,58 @@
...
+		"0xdead",
+		"pending"
+    ],
+    "id": 1
+}
+
+HTTP 200
+[Asserts]
+jsonpath "$.error.message" equals "Contract not found"
+jsonpath "$.result" not exists
```
Comment (glihm) [inline examples/rpc/starknet/starknet_getStorageAt.hurl:58]:
```suggestion
jsonpath "$.result" not exists

```

**Review by glihm — COMMENTED** [review]
Great! Except the two missing new lines, I don't see much to change. Running tests with StarknetJS `5.24.3` it's working nice.

**Review by glihm — APPROVED** [review]

**Pull request merged**